### PR TITLE
Add send to the first channel of server

### PR DESCRIPTION
**Pull request by Serhann**
**Please describe the changes this PR makes and why it should be merged:**
This code alternative of `.defaultChannel`

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by Dev-Yukine** [comment]
uhm if i remember correct collections are not sorted like the actuall channel order or? 🤔

**Comment by linuxgemini** [comment]
Channel Collections in general are not sorted by channel order, got me thinking as well.

**Comment by Dev-Yukine** [comment]
i would add a sort function for that something like this
`<channelsCollection>.sort((a, b) => a.position - b.position);`

**Comment by devsnek** [comment]
this is not a safe addition due to its abstract nature and should be closed.

**Comment by devsnek** [comment]
`channels.sort((a, b) => a.position - b.position || a.id - b.id);

**Comment by linuxgemini** [comment]
`msg.guild` must be `member.guild` by the way.

File: docs/examples/greeting.js [hunk 1]
```diff
@@ -19,7 +19,14 @@ client.on('ready', () => {
 
 // Create an event listener for new guild members
 client.on('guildMemberAdd', member => {
-  // Send the message to a designated channel on a server:
+  // Send the message to a first channel on a server:
+  let firstTextCh = msg.guild.channels
+    .filter(c => c.type === "text")
+    .filter(c => c.permissionsFor(c.guild.member(client.user)).has("SEND_MESSAGES"))
+    .first();
```
Comment (Dev-Yukine) [inline docs/examples/greeting.js]: use instead `member.guild.channels.filter(c => c.type === "text" && c.permissionsFor(msg.guild.me).has("SEND_MESSAGES")).sort((a, b) => a.position - b.position || a.id - b.id).first()`

**Comment by Dev-Yukine** [comment]
you still miss that collection are not sorted like the actuall channel order btw.

File: docs/examples/greeting.js [hunk 2]
```diff
@@ -19,7 +19,14 @@ client.on('ready', () => {
 
 // Create an event listener for new guild members
 client.on('guildMemberAdd', member => {
-  // Send the message to a designated channel on a server:
+  // Send the message to a first channel on a server:
```
Comment (PgBiel) [inline docs/examples/greeting.js]: "to a first channel on a server"
you mean, "to **the** first channel on a server"

**Comment by linuxgemini** [comment]
Also, due to the nature of the code, the same message has the possibility to being sent twice.

    // Create an event listener for new guild members
    client.on('guildMemberAdd', member => {
      // Try finding a designated channel on a server:
      const channel = member.guild.channels.find('name', 'member-log');

      // Find the very first channel if the channel wasn't found on this server
      if (!channel) {
        const channel = member.guild.channels
          .filter(c => c.type === "text")
          .filter(c => c.permissionsFor(c.guild.member(client.user)).has("SEND_MESSAGES"))
          .sort((a, b) => a.position - b.position)
          .first();
      }
      
      // Do nothing if the is no channel to send the message
      if (!channel) return;
      
      // Send the message, mentioning the member
      channel.send(`Welcome to the server, ${member}`);
    });

**Comment by iCrawl** [comment]
@linuxgemini This is an example to indicate that both methods can be used, but should not be used at the same time. It's redundant to do those checks.

**Comment by iCrawl** [comment]
As others have mentioned this will not work the way you expect it to work since you filter incorrectly.

Adding to that I feel like this should not be an official "example" because its a rather touchy subject with an outcome you might not expect or want to have.

**Pull request closed without merging**